docs/src/components/productX/XTreeViewDemo.tsx

```diff
@@ -1,261 +1,276 @@
-    if (extension.match(/(jpg|jpeg|png)/)) {
-      return <PhotoOutlined />;
-    }
-    if (extension === 'pdf') {
-      return <PictureAsPdfOutlined />;
-    }
-    if (extension === 'mp4') {
-      return <VideocamOutlined />;
-    }
-    if (extension === 'mkv') {
-      return <FourKOutlined />;
-    }
-    return (
-      <Box
-        sx={{
-          width: 6,
-          height: 6,
-          borderRadius: '50%',
-          bgcolor: 'warning.main',
-          display: 'inline-block',
-          verticalAlign: 'middle',
-          zIndex: 1,
-        }}
-      />
-    );
-  };
+  return <AnimatedCollapse style={style} {...props} />;
+}
+
+interface CustomLabelProps {
+  children: React.ReactNode;
+  icon?: React.ElementType;
+  expandable?: boolean;
+}
 
+function CustomLabel({ icon: Icon, expandable, children, ...other }: CustomLabelProps) {
   return (
-    /* @ts-ignore -- Key event is handled by the TreeView */
-    <Box
-      className={clsx(className, classes.root, {
-        [classes.expanded]: expanded,
-        [classes.selected]: selected,
-        [classes.focused]: focused,
-        [classes.disabled]: disabled,
-        'CustomContent-last': lastNestedChild,
-      })}
-      onClick={handleExpansionClick}
-      onMouseDown={handleMouseDown}
-      ref={ref as React.Ref<HTMLButtonElement>}
+    <TreeItem2Label
+      {...other}
       sx={{
-        border: 'none',
-        borderRadius: '5px',
-        textAlign: 'left',
-        position: 'relative',
-        zIndex: 1,
-        '& svg': {
-          fontSize: 18,
-          color: nodeId.startsWith('root') ? 'primary.main' : 'grey.600',
-        },
-        '&:not(.CustomContent-last)': {
-          '& svg': {
-            '&:first-of-type': {
-              fontSize: 14,
-              color: 'primary.main',
-            },
-          },
-        },
+        display: 'flex',
+        alignItems: 'center',
       }}
     >
-      {icon}
-      {lastNestedChild && renderExtension()}
-      {!lastNestedChild && (expanded ? <FolderOpenRounded /> : <FolderRounded />)}
+      {Icon && (
+        <Box
+          component={Icon}
+          className="labelIcon"
+          sx={(theme) => ({
+            mr: 1,
+            fontSize: '1rem',
+            color: expandable ? theme.palette.primary.main : theme.palette.grey[600],
+          })}
+        />
+      )}
+
       <Typography
-        onClick={handleSelectionClick}
-        component="div"
-        className={classes.label}
-        noWrap
-        sx={{
-          '&&': {
-            color: lastNestedChild ? 'text.secondary' : 'text.primary',
-            fontWeight: lastNestedChild ? 400 : 500,
-          },
-        }}
+        sx={(theme) => ({
+          fontWeight: expandable
+            ? theme.typography.fontWeightMedium
+            : theme.typography.fontWeightRegular,
+          color: expandable ? theme.palette.text.primary : theme.palette.grey[600],
```

Comment:
```suggestion
          color: expandable ? theme.palette.text.primary : theme.palette.text.secondary,
```

So we ensure contrast